### DISCUSSION TITLE
Live preview without rendering?

### DISCUSSION THREAD

**Discussion by kvten** [quarto-dev/quarto-cli#3067] [Q&A] [answered]
Is there an option to have live preview and/or see changes of the quarto document you are working on without having to re-render the output file?

If this doesn't exist yet and is feasible. Can it be added to suggestions list? Either RStudio or VScode but preferrably VScode.

Thanks!

**Comment by mcanouil** [comment, marked as answer]
This is already the case or kind of unless `--no-watch-inputs` is used.
Although, it's not really "live" since, it is simply re-rendered on file save.

EDIT: for VSCode, there is an option to switch on
<img width="512" alt="image" src="https://user-images.githubusercontent.com/8896044/198583685-4189589e-f832-4d6c-8249-bbe2866ad8a3.png">

EDIT2: Switching on Windows to get a screenshot of the latest RStudio
![image](https://user-images.githubusercontent.com/8896044/198585497-6a388c7d-da1a-49aa-8ee1-5b0af70ba504.png)

**Reply by dragonstyle** [reply]
There is a `render on save` option in VSCode that also may be helpful - that will perhaps give you the experience you'd like (make changes, save, see changes in preview).

edit: To clarify - the Quarto extension in VSCode has this option
![Screen Shot 2022-10-28 at 8 08 41 AM](https://user-images.githubusercontent.com/261654/198583646-9f39634e-58f2-4e74-9e25-f6b39f6ec666.png)

**Reply by mcanouil** [reply]
I was going to add the screenshot ^^
<img width="512" alt="image" src="https://user-images.githubusercontent.com/8896044/198583607-539f67a1-dbee-42fa-bc6c-529683b3ae0f.png">

**Reply by xmaxrayx** [reply]
instead of preview on RAM let's ruin SSD lifespan.....

**Reply by mcanouil** [reply]
> instead of preview on RAM let's ruin SSD lifespan.....

@xmaxrayx what's the point of your comment? We expect comment to be constructive at least.
And here you are commenting on a two years old thread for no obvious or informative reason.

> [!IMPORTANT]
> We appreciate your interest in the Quarto project, but the content above violates our [Code of Conduct](https://github.com/quarto-dev/quarto-cli?tab=coc-ov-file#readme).
>
> As a member of the Quarto community, you are expected to follow our Code of Conduct, which outlines the standards of behavior and communication that we value. The Code of Conduct applies to all online interactions within the Quarto project, and it is enforced by the Quarto team. By participating in the Quarto community, you agree to abide by the Code of Conduct and respect the rights and dignity of others. You can find the Code of Conduct at <https://github.com/quarto-dev/quarto-cli?tab=coc-ov-file#readme>.
> 
> Thank you for your cooperation and contribution. 🙏

**Comment by xmaxrayx** [comment]
No, because devs choose to compile it outside vscode with that CLI thing.

**Reply by mcanouil** [reply]
The user asked within the IDE such as RStudio and VSCode.

The CLI has flags for that.

Additionally, to what exactly are you replying to "no"? It seems off topic.
